crypto/keyring/record.go

```diff
@@ -81,19 +80,24 @@ func (re Record) GetType() KeyType {
 	return 0
 }
 
-func (re Record) extractPrivKeyFromLocalInfo() (cryptotypes.PrivKey, error) {
+func (re *Record) extractPrivKeyFromLocal() (cryptotypes.PrivKey, error) {
+	
 	local := re.GetLocal()
-
+	fmt.Println("extractPrivKeyFromLocal local PrivKey any", local.PrivKey)
+	"S�local PrivKey any &Any{TypeUrl:/cosmos.crypto.secp256k1.PrivKey,Value:[10 32 60 192 254 115 242 129 186 183 124 20 160 13 47 202 179 92 24 116 152 216 145 44 66 161 255 183 157 144 113 154 45 201],XXX_unrecognized:[]}"
+	fmt.Println("extractPrivKeyFromLocal local PubKeyType", local.PubKeyType)
+	"secp256k1"
+	
 	switch {
 	case local != nil:
-		fmt.Println("extractPrivKeyFromLocalInfo PrivKey", local.PrivKey)
-		privKey, ok := local.PrivKey.GetCachedValue().(cryptotypes.PrivKey)
+		anyPrivKey := local.PrivKey
+		privKey, ok := anyPrivKey.GetCachedValue().(cryptotypes.PrivKey)
 		if !ok {
-			return nil, fmt.Errorf("unable to cast to cryptotypes.PrivKey")
+			return nil, fmt.Errorf("unable to unpack private key")
```

Comment:
@amaurym  I have printed out all fields of local on lines 86-89. Just wonder why I get an this error on line 96 `unable to unpack private key`.